### PR TITLE
Improve tests and create acceptance test for UsageKind

### PR DESCRIPTION
**Pull request by polskikiel**
Confirm these statements before you submit your pull request:

- [x] I have read and submitted the required [Contributor Licence Agreements](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md#agreements-and-licenses).
- [x] This pull request follows the contributing guidelines, recommended Git workflow, and templates.
- [x] I have tested my changes.
- [x] I have updated the relevant documentation.
---

Changes proposed in this pull request:

- Acceptance test for UsageKind in ui-api
- Improvements of the unit tests in ui-api
- Rbac roles for acceptance test

### CODE REVIEW

File: components/ui-api-layer/README.md [hunk 1]
```diff
@@ -97,3 +97,15 @@ KUBE_CONFIG=/Users/{your_username}/.kube/config go test ./... -tags=acceptance
 ### Verify the code
 
 To check if the code is correct and you can push it, run the `before-commit.sh` script. It builds the application, runs tests, checks the status of the vendored libraries, runs the static code analysis, and ensures that the formatting of the code is correct.
+
+
+### Custom UsageKind
+
+In order to specify a new custom UsageKind, you have to adjust RBAC roles of the UI API Layer. UsageKind contains a reference to the resource. You have to give UI API Layer `get`, `list` and `watch` permissions to this resource.
+If UsageKind references to the Pods, then you have to add below permission to the UI API Layer RBAC [roles](../../resources/core/charts/ui-api/templates/cluster-role.yaml):
```
Comment (klaudiagrz) [inline components/ui-api-layer/README.md]: 1. ~references~ -> `refers`
2. ~then~
3. ~have to~ -> `must`
4. ~below~

Add this sentence at the end:
`See the following example:`

File: components/ui-api-layer/internal/domain/servicecatalog/usage_kind_resolver_test.go [hunk 2]
```diff
@@ -5,71 +5,66 @@ import (
...
 	resolver := newUsageKindResolver(svc)
+
 	// WHEN
-	resp, err := resolver.ListServiceUsageKindResources(context.Background(), "fix-A", "default")
-	require.NoError(t, err)
+	_, err := resolver.ListServiceUsageKindResources(context.Background(), "fix-A", fixUsageKindResourceNamespace())
 
 	// THEN
-	assert.Equal(t, []gqlschema.UsageKindResource{}, resp)
+	require.NoError(t, err)
```
Comment (aszecowka) [inline components/ui-api-layer/internal/domain/servicecatalog/usage_kind_resolver_test.go]: again, why ignoring response?

File: components/ui-api-layer/internal/domain/servicecatalog/usage_kind_service_test.go [hunk 3]
```diff
@@ -48,30 +48,26 @@ func TestUsageKindService_List_Empty(t *testing.T) {
 	require.NoError(t, err)
 
 	// THEN
-	assert.Len(t, result, 0)
+	assert.Empty(t, result)
```
Comment (aszecowka) [inline components/ui-api-layer/internal/domain/servicecatalog/usage_kind_service_test.go]: TestUsageKindService_List_Success does not test too much, there is the only assertion about the length of a returned slice.

File: tests/ui-api-layer-acceptance-tests/domain/servicecatalog/usage_kind_test.go [hunk 4]
```diff
@@ -0,0 +1,235 @@
+// +build acceptance
```
[the remark this answers is not in the document]
Comment (polskikiel) [reply tests/ui-api-layer-acceptance-tests/domain/servicecatalog/usage_kind_test.go]: Done...

File: components/ui-api-layer/internal/domain/servicecatalog/usage_kind_service_test.go (hunk 3 above)
Comment (polskikiel) [reply components/ui-api-layer/internal/domain/servicecatalog/usage_kind_service_test.go]: I changed `TestUsageKindService_List_Success` assertion about the length of a returned slice. Now i check result using assert.Equal().

File: components/ui-api-layer/internal/domain/servicecatalog/usage_kind_resolver_test.go [hunk 5]
```diff
@@ -5,71 +5,66 @@ import (
...
 	resolver := newUsageKindResolver(svc)
+
 	// WHEN
-	resp, err := resolver.ListUsageKinds(context.Background(), nil, nil)
-	require.NoError(t, err)
+	_, err := resolver.ListUsageKinds(context.Background(), nil, nil)
 
 	// THEN
-	assert.Equal(t, resp, fixUsageKindsGQL())
+	require.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (polskikiel) [reply components/ui-api-layer/internal/domain/servicecatalog/usage_kind_resolver_test.go]: Ok, I restored this change. But like below i test almost whole logic using above mock.

File: tests/ui-api-layer-acceptance-tests/domain/servicecatalog/usage_kind_test.go [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+
+	var usageKindResourcesResponse usageKindResourcesResponse
+	t.Log("Querying for usageKindResources...")
+	err = c.Do(fixUsageKindsResourcesQuery(), &usageKindResourcesResponse)
+
+	require.NoError(t, err)
+	assert.NotEmpty(t, usageKindResourcesResponse.UsageKindResources)
+
+	t.Log("Creating resource for UsageKind with owner reference...")
+	_, err = deployClient.Deployments(usageKindNamespace).Create(fixDeploymentWithOwnerReference())
```
[the remark this answers is not in the document]
Comment (polskikiel) [reply tests/ui-api-layer-acceptance-tests/domain/servicecatalog/usage_kind_test.go]: Ok, good point.

File: tests/ui-api-layer-acceptance-tests/domain/servicecatalog/usage_kind_test.go [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+	require.NoError(t, err)
+
+	deployClient, err := v1beta1.NewForConfig(cfg)
+	require.NoError(t, err)
+
+	t.Log("Creating namespace...")
+	_, err = client.Namespaces().Create(fixNamespace(usageKindNamespace))
+	require.NoError(t, err)
+
+	t.Log("Creating UsageKind...")
```
[the remark this answers is not in the document]
Comment (polskikiel) [reply tests/ui-api-layer-acceptance-tests/domain/servicecatalog/usage_kind_test.go]: Ok, I checked it and we get 2 default UsageKinds while kyma is installing. I can use them in this test but I will discuss it with the rest of the gophers.

File: tests/ui-api-layer-acceptance-tests/domain/servicecatalog/usage_kind_test.go [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+	_, err = bucClient.ServicecatalogV1alpha1().UsageKinds().Create(fixUsageKind())
+	require.NoError(t, err)
+
+	defer func() {
+		t.Log("Deleting namespace...")
+		err = client.Namespaces().Delete(usageKindNamespace, &metav1.DeleteOptions{})
+		require.NoError(t, err)
```
Comment (piotrmiskiewicz) [inline tests/ui-api-layer-acceptance-tests/domain/servicecatalog/usage_kind_test.go]: change to assert to not block defer processing

File: tests/ui-api-layer-acceptance-tests/domain/servicecatalog/usage_kind_test.go [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+		err = bucClient.ServicecatalogV1alpha1().UsageKinds().Delete(usageKindName, &metav1.DeleteOptions{})
+		require.NoError(t, err)
+	}()
+
+	var usageKindsResponse usageKindsResponse
+	t.Log("Querying for usageKinds...")
+	err = c.Do(fixUsageKindsQuery(), &usageKindsResponse)
+
+	require.NoError(t, err)
+	assert.Contains(t, usageKindsResponse.UsageKinds)
```
Comment (piotrmiskiewicz) [inline tests/ui-api-layer-acceptance-tests/domain/servicecatalog/usage_kind_test.go]: fix the contains - should have 3 args

**Pull request merged**